rpcserver.go

```diff
@@ -2330,6 +2346,34 @@ func (r *rpcServer) AssetLeaves(ctx context.Context,
 	return resp, nil
 }
 
+// unmarshalOutpoint unmarshals an outpoint from a string received via RPC.
+func UnmarshalOutpoint(outpoint string) (*wire.OutPoint, error) {
+	parts := strings.Split(outpoint, ":")
+	if len(parts) != 2 {
+		return nil, errors.New("outpoint should be of form txid:index")
+	}
+
+	txidStr := parts[0]
+	if hex.DecodedLen(len(txidStr)) != chainhash.HashSize {
+		return nil, fmt.Errorf("invalid hex-encoded txid %v", txidStr)
+	}
+
+	txid, err := chainhash.NewHashFromStr(txidStr)
+	if err != nil {
+		return nil, err
+	}
+
+	outputIndex, err := strconv.Atoi(parts[1])
+	if err != nil {
+		return nil, fmt.Errorf("invalid output index: %v", err)
+	}
+
+	return &wire.OutPoint{
+		Hash:  *txid,
+		Index: uint32(outputIndex),
+	}, nil
+}
+
```

Comment:
I think this can be replaced by `NewOutPointFromString` from `btcd.wire`.

Reply:
Good catch! I'm wondering if we care that this would move the `btcd` depedency from a tagged version to master / a specific commit?

Reply:
Tracked in #362.